pandas/tests/frame/test_arithmetic.py

```diff
@@ -99,6 +99,44 @@ def test_df_flex_cmp_constant_return_types_empty(self, opname):
 # Arithmetic
 
 class TestFrameFlexArithmetic(object):
+    # TODO: tests for other arithmetic ops
+    def test_df_add_2d_array_rowlike_broadcasts(self):
+        # GH#
+        arr = np.arange(6).reshape(3, 2)
+        df = pd.DataFrame(arr, columns=[True, False], index=['A', 'B', 'C'])
+
+        rowlike = arr[[1], :]  # shape --> (1, ncols)
+        expected = pd.DataFrame([[2, 4],
```

Comment:
can you assert on the shape of rowlike here (like your comment but more explict)